angular/src/providers/nav-controller.ts

```diff
@@ -202,7 +202,7 @@ export class NavController {
         urlTree.fragment = options.fragment;
       }
 
-      return this.router!.navigateByUrl(urlTree);
+      return this.router!.navigateByUrl(urlTree, options);
```

Comment:
```suggestion
      /**
       * `navigateByUrl` will still apply `NavigationExtras` properties
       * that do not modify the url, such as `replaceUrl` which is why 
       * `options` is passed in here.
       */
      return this.router!.navigateByUrl(urlTree, options);
```

Reply:
Since this caveat about navigateByUrl is a little confusing/hard to remember, let's add an extra comment that just makes it very clear what we are doing and why.

Reply:
@liamdebeasi Great, thanks!

Reply:
👍 Also you'll need to commit this yourself since I do not have permission to write to your branch.